src/inmanta/ast/__init__.py

```diff
@@ -382,6 +387,9 @@ def get_type(self, typ: LocatableString) -> "Type":
             else:
                 raise TypeNotFoundException(typ, self)
         elif name in self.primitives:
+            if name == "number":
+                msg = "Type 'number' is deprecated, use 'float' or 'int' instead (%s)" % str(typ.location)
+                warnings.warn(TypeDeprecationWarning(msg))
```

Comment:
Did you verify that you only get the warning once, no matter how many number types you use?

Reply:
No you get it for each use of the number type. But isn't that better? I would like to see all the usages of the number type with where they are used at once so that I can make all the changes without needing to run the code multiple times.

Reply:
I think the solutions team will disagree very strongly. They will not be able to update immediately after this change, and they may well have hundreds of usages of this type. This would pollute their logs incredibly.

Reply:
But so then no need to have an indication of the line where it is used? Just one warning saying that the type is still present?

Reply:
Ah, I think `warnings.warn` shows each warning only once, but the message differs here for each of them because of the location information. Perhaps we can ask the solutions team how they would prefer to be warned about this?

Reply:
> But so then no need to have an indication of the line where it is used? Just one warning saying that the type is still present?

I'm honestly not sure. I agree that pointing to the line is best, but I don't immediately see how to achieve both goals.

Reply:
It's seems even when removing the location a lot of warnings are produced. I'll need to look how to fix this